### DISCUSSION TITLE
[Help wanted] Unable to add PPA for Ubuntu Kinetic

### DISCUSSION THREAD

**Discussion by Dragon1573** [wslutilities/wslu#254] [Q&A] [answered]
> This question is related to another question on [Launchpad Answers](https://answers.launchpad.net/ubuntu/+source/wslu/+question/704175).
>
> I asked the question there as [Installation - WSLu Chinese Wiki](https://wslutiliti.es/wslu/zh-CN/install.html) instructed, but it seems that I asked in a wrong place ...
>
> ![image](https://user-images.githubusercontent.com/49941141/208886772-094f3139-2053-413f-aade-45cbc1e9cbdd.png)

## Target

Install `wslu 4.0.0` on Ubuntu 22.10 Kinetic.

## Environments

- Windows 11 Professional 22H2
- Windows Subsystem Linux (version 2)
- Ubuntu 22.10 Kinetic Kudu

## Summary

According to installation guide, `wslu` is not preinstalled from Ubuntu 22.04 LTS (Jammy).

![image](https://user-images.githubusercontent.com/49941141/208885187-9d7a198d-9f42-4ce4-834f-5e685d79cf8e.png)

I add the PPA repository to my Ubuntu Kinetic and an error occurs.

```text
13:34:07 ~/Documents 2ms $ sudo add-apt-repository ppa:wslutilities/wslu
[sudo] dragon1573 的密码：
Repository: 'deb https://ppa.launchpadcontent.net/wslutilities/wslu/ubuntu/ kinetic main'
Description:
Official repository for wslu packages
More info: https://launchpad.net/~wslutilities/+archive/ubuntu/wslu
Adding repository.
Press [ENTER] to continue or Ctrl-c to cancel.
Found existing deb entry in /etc/apt/sources.list.d/wslutilities-ubuntu-wslu-kinetic.list
Adding deb entry to /etc/apt/sources.list.d/wslutilities-ubuntu-wslu-kinetic.list
Found existing deb-src entry in /etc/apt/sources.list.d/wslutilities-ubuntu-wslu-kinetic.list
Adding disabled deb-src entry to /etc/apt/sources.list.d/wslutilities-ubuntu-wslu-kinetic.list
Adding key to /etc/apt/trusted.gpg.d/wslutilities-ubuntu-wslu.gpg with fingerprint 254F460F2970E18123046570C1D0E7E6AB4095D6
命中:1 http://repo.huaweicloud.com/ubuntu kinetic InRelease
命中:2 http://repo.huaweicloud.com/ubuntu kinetic-updates InRelease
命中:3 http://repo.huaweicloud.com/ubuntu kinetic-backports InRelease
命中:4 http://repo.huaweicloud.com/ubuntu kinetic-security InRelease
忽略:5 https://ppa.launchpadcontent.net/wslutilities/wslu/ubuntu kinetic InRelease
错误:6 https://ppa.launchpadcontent.net/wslutilities/wslu/ubuntu kinetic Release
  404 Not Found [IP: 185.125.190.52 443]
正在读取软件包列表... 完成
E: 仓库 “https://ppa.launchpadcontent.net/wslutilities/wslu/ubuntu kinetic Release” 没有 Release 文件。
N: 无法安全地用该源进行更新，所以默认禁用该源。
N: 参见 apt-secure(8) 手册以了解仓库创建和用户配置方面的细节。
```

I check the PPA repository and could not find available version for Kinetic. I find an available `wslu` in official `kinetic` source but it only provides `3.2.3-0ubuntu3` version instead of `4.0.0`. This version can't provide some parameters as [the manual page](https://wslutiliti.es/wslu/man/wslusc.html).

```text
13:38:47 ~/Documents 2ms $ sudo apt search '^wslu$'
正在排序... 完成
全文搜索... 完成
wslu/kinetic 3.2.3-0ubuntu3 amd64
  collection of utilities for the Windows 10 Linux Subsystem
```

**Comment by car-meows** [comment, marked as answer]
Hi,

Thank you for the feedback! I just freed from the work so I can recently back on working on `wslu`. Currently we do not have a build for Ubuntu Kinetic, but I will do one before end of this year. Thank you for your understanding.

**Reply by Dragon1573** [reply]
Okay, @patrick330602. `wslu` is a powerful and useful tool for me, especially `wslview` and `wslusc`! Thank you for your masterpiece! 🙏🏼

I'm waiting for your new release! ❤️

## Additional Informations

My WSL2 instance is configured with the following file, so maybe I'll rolling to **Lunar Lobster** once it's released (according to my search engine, it said that the release date will be **April 20, 2023**). Could you please also provide a package for it?

```text
00:03:51 ~/Documents 1ms $ cat /etc/update-manager/release-upgrades
# Default behavior for the release upgrader.

[DEFAULT]
# Default prompting and upgrade behavior, valid options:
#
#  never  - Never check for, or allow upgrading to, a new release.
#  normal - Check to see if a new release is available.  If more than one new
#           release is found, the release upgrader will attempt to upgrade to
#           the supported release that immediately succeeds the
#           currently-running release.
#  lts    - Check to see if a new LTS release is available.  The upgrader
#           will attempt to upgrade to the first LTS release available after
#           the currently-running one.  Note that if this option is used and
#           the currently-running release is not itself an LTS release the
#           upgrader will assume prompt was meant to be normal.
Prompt=normal
```

I'm currently using the "Ubuntu official" version of `wslu` as described above.

1. Could you please tell me the difference between them (or they are the same, just from different publisher)?
2. Does the WSLu Wiki intended for PPA version instead of the "Ubuntu official" one?
3. Shall I uninstall the current `wslu` and reinstall the PPA one once its available on Kinetic (and also Lunar)?

**Reply by car-meows** [reply]
@Dragon1573 For the first question, the official version is used to be maintained by me while I was still working in Canonical, the company behind Ubuntu. The PPA version at the time served as the location that people could get the latest version of wslu before getting into the official repository. But:

1. Process of submission of new version of wslu to official repository is too slow;
2. I left the company.

Thus, technically, I no longer own the rights for the official repository versions of wslu; they are now supposed to be maintained by Ubuntu WSL team. However, as I still own the PPA, it is the version right now used for Ubuntu.

For the second question, the wiki always reflect the latest version of wslu so yes, it is PPA version.

For the third question, yes, you should better remove it and install via PPA. **The latest version should be available now for Bionic, Focal, Jammy, Kinetic and Lunar on PPA.**

**Reply by Dragon1573** [reply]
Thank you! Successfully migrated to PPA version ❤️

```text
16:30:01 ~/Documents 1ms $ apt info wslu
Package: wslu
Version: 4.1.0-1ubuntu1~22.10
Priority: optional
Section: admin
Maintainer: Patrick Wu <me@patrickwu.space>
Installed-Size: 303 kB
Provides: www-browser
Depends: desktop-file-utils, bc, psmisc
Recommends: pulseaudio-utils, x11-utils
Suggests: imagemagick
Download-Size: 86.3 kB
APT-Manual-Installed: yes
APT-Sources: https://ppa.launchpadcontent.net/wslutilities/wslu/ubuntu kinetic/main amd64 Packages
Description: collection of utilities for the Linux Subsystem for Windows
 This is a collection of utilities for the Linux Subsystem for Windows
 (WSL), such as for converting Linux paths to a Windows paths or creating
 Linux application shortcuts on the Windows Desktop.
 .
 Requires Windows 10 Creators Update and higher.

N: 有 1 条附加记录。请加上 ‘-a’ 参数来查看它们

16:32:08 ~/Documents 1ms $ wslusc --<TAB>
--env                          (add a custom environment to your command)  --name        (add a different name other than the command name)
--gui                 (create a shortcut that links to a GUI application)  --native  (create a shortcut that uses the system builtin 'wslg)
--help                                              (print a simple help)  --shortcut-debug             (debug the generated shortcut file)
--icon  (add an icon to the Windows Shortcut. Supports . ico/. png/. svg)  --smart-icon             (This enables the smart icon detection)
--interactive                           (interactive mode. CONFIGURATION)  --version                                (print current version)
```
